RELEASE.md

```diff
@@ -47,26 +48,34 @@ See the next section for details on cutting an individual release.
 The quick and easy way to create a GitHub issue using the following template and follow the instructions from the issue itself.
 If something is not clear, you can get back to this document to learn more about the process.
 
+> **Note:** When making a branch while working on the following checklist, don't create branch with `release-` prefix, because such branch is protected and can't get additional commits pushed upon branch creation.
+
 ````markdown
 ### Publish the release candidate
 
 - [ ] Begin drafting the [release notes](https://github.com/grafana/mimir/blob/main/RELEASE.md#write-release-notes-document)
-  - Write release notes to `main`, then cherry pick them into the release branch
-  - Don't block on it to publish the release candidate
+  - Create the release note PR targeting the main branch
+  - This step shouldn't block from publishing release candidate
+  - After the release note PR is merged (which usually happen after RC is published), cherry-pick them into the release branch
 - [ ] Wait for any open PR we want to get merged before cutting the release candidate
+  - We shouldn't wait the open PRs beyond the scheduled release date
   - [ ] Eventually open a PR for every experimental feature we want to promote to stable
   - [ ] Eventually open a PR to remove any deprecated feature or configuration option that should be removed in this release
 - [ ] Update `CHANGELOG.md`
   - [ ] Run `./tools/release/check-changelog.sh LAST-RELEASE-TAG...main` and add missing PRs to CHANGELOG
   - [ ] Ensure CHANGELOG entries are [sorted by type](https://github.com/grafana/mimir/blob/main/docs/internal/contributing/README.md#changelog)
   - [ ] Add a new section for the new release so that `## main / unreleased` is blank and at the top. The new section should say `## x.y.0-rc.0`.
 - [ ] Run `./tools/release/notify-changelog-cut.sh`
-- [ ] Run `make mixin-screenshots`
+- [ ] Generate mixin screenshots
+  - Update config in `operations/mimir-mixin-tools/screenshots/.config` and `operations/mimir-mixin-tools/serve/.config`
+  - Use dev `MIMIR_NAMESPACE` that has more metrics such as `cortex-dev-01` so that the generated dashboard will not be too empty
+  - Use api-keys generated from [grafana dev environment](https://grafana-dev.com/orgs/raintank/api-keys) to set the `DATASOURCE_PASSWORD`
```

Comment:
This refers to Grafana Labs specific environment, which is not useful in a public document.